File: Lecture1/SoccerManager/Player.cpp
#include <stdlib.h>
#include "Player.h"

//constructor
Player::Player():
    Person("J. Doe",18),
    strength(1),
    shots(1),
    motivation(1),
    goals(0) {}
Player::Player(string n,int a,int st, int sh, int mo):
    Person(n,a),
    strength(st),
    shots(sh),
    motivation(mo),
    goals(0) {}
//get set function
int Player::getGoals()
{
    return goals;
}
//player function
void Player::addGoal()
{
    goals++;
}

int Player::shootsOnGoal()
{
    shots=max(1,min(10,shots+rand()%3-1));
    int ret=max(1,min(10,shots+rand()%3-1));
    return ret;
}
int Player::getStrength()
{
    return strength;
}
int Player::getMotivation()
{
    return motivation;
}

File: Lecture5/Python/SoccerManager/SoccerManager/Player.py
import random

from Person import *


class Player(Person):
    __strength = 1
    __shots = 1
    __motivation = 1
    __goals = 0

    def __init__(self, name, age, st, sh, mo):
        Person.__init__(self, name, age)
        self.__strength = st
        self.__shots = sh
        self.__motivation = mo

    def print_info(self):
        print("PLAYER INFO:")
        print("Name: "+self.get_name())
        print("Age: " + str(self.get_age()))
        print("Stats: ["+str(self.__strength)+","+str(self.__shots)+","+str(self.__motivation)+"]")

    def get_goals(self) -> int:
        return self.__goals

    def add_goal(self):
        self.__goals += 1

    def shoots_on_goal(self) -> int:
        self.__shots = max(1, min(10, self.__shots+random.randint(-1, 1)))
        ret = max(1, min(10, self.__shots+random.randint(-1, 1)))
        return ret

    def get_strength(self) -> int:
        return self.__strength

    def get_motivation(self) -> int:
        return self.__motivation

